docs/Deployment.md

```diff
@@ -0,0 +1,663 @@
+   instances.
+
+Both of these are standard components that are not specific to Certificate
+Transparency, and so will only be covered briefly.
+
+Finally, an optional (but highly recommended) component for a deployed Log is
+some kind of monitoring for the operational status of the Log components.
+
+
+Manual Setup
+============
+
+ - [Required Configuration](#required-configuration)
+    - [Private Key Generation](#private-key-generation)
+    - [CA Certificates](#ca-certificates)
+ - [Optional Configuration](#optional-configuration)
+ - [etcd Setup](#etcd-setup)
+ - [Monitoring Setup](#monitoring-setup)
+    - [GCM](#gcm)
+    - [Prometheus](#prometheus)
+ - [Front-End Setup](#front-end-setup)
+ - [Standalone Setup](#standalone-setup)
+
+Required Configuration
+----------------------
+
+The `ct-server` binary (from `cpp/server/`) is the core of the running CT Log
+system.  Multiple distributed instances of `ct-server` should be run in
+parallel to allow scaling and to ensure resilience:
+
+ - Running multiple `ct-server` instances allows the Log to scale with traffic
+   levels, by adjusting the number of instances.
+ - Running `ct-server` instances in distinct locations reduces the chance of a
+   single external event affecting all instances simultaneously.  (In terms of
+   cloud computing providers, this means that instances should be run in
+   different zones/regions/availability zones.)
+
+
+Each `ct-server` instance needs to be configured with information about how it
+should run and its other dependencies.  The most important configuration options
+are:
+
+ - `--key=<pemfile>` specifies the private key that the Log will use to perform
+   cryptographic signing operations.  This key should not be password
+   protected, but needs to be [kept carefully](Operation.md#key-management)
+   to prevent Log failure.  See [below](#private-key-generation) for
+   tips on how to generate a private key.
+ - `--trusted_cert_file=<pemfile>` specifies the set of CA certificates that
+   the Log accepts as trust roots for logged certificates.  See
+   [below](#ca-certificates) for hints on possible ways to build this set.
+ - `--leveldb_db=<file>.ldb` specifies the
+   [LevelDB](https://github.com/google/leveldb) storage directory used to
+   store the certificate data for the Log.  This needs to be stored on a
+   persistent disk, and needs to be in a separate location for each
+   `ct-server` instance.
+ - `--etcd_servers=<host>:<port>,<host>:<port>,...` specifies the set of `etcd`
+   servers that should be used for data synchronization and replication.  See
+   [below](#etcd-setup) for information about `etcd` setup.
+
+Other configuration options that need to be set to non-default values are:
+
+ - `--server=<hostname>` identifies the local machine; this is used to
+   distinguish different clustered Log instances when synchronizing, so needs to
+   be distinct for each `ct-server` instance (i.e. cannot be `localhost`).
```

Comment:
also needs to be resolvable & routable from the other instances (bit obvious, perhaps)